### ISSUE TITLE
Feature suggestion: Fetching physical size data from monitors

### ISSUE THREAD

**Issue by hhannine** [rr-/screeninfo#23]
First of all, thank you kindly for creating this useful tool! I'm using it in [Superpaper](https://github.com/hhannine/Superpaper/). I'm looking into automating some of the config out of the users hands and this would need the automatic detection of the display height and width in millimeters, inches etc.; or the detection of the monitor pixel density PPI directly.

Would you think that extension of screeninfo to report these additional data would be appropriate? If it could be considered there's of course the question of how to get at the data. On linux at least xrandr seems to report the width and height in millimeters. On the other hand, on Windows I haven't been able to find any easy way to get at it and with my limited understanding one way would be to read cached EDID data out of the registry.

**Comment by hhannine** [comment]
I found a reference to [GetDeviceCaps](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/nf-wingdi-getdevicecaps) in [here](https://social.msdn.microsoft.com/Forums/vstudio/en-US/fdb23220-9845-43f2-a4ed-0223caf64368/query-monitor-physical-size?forum=vcgeneral), looks a bit more promising than having to read it yourself from somewhere.

It also looks like after a glance that the xrandr enumerator already is aware of the millimeter sizes of the monitors, its just not passed on to the output?

**Comment by rr-** [comment]
Sounds good, I'll look into it

**Comment by hhannine** [comment]
Great! Once you lay down the functionality for xrandr maybe I can help figuring out it on Windows.

I was able to get some reasonable values with GetDeviceCaps with the following addition to the callback() method in the windows enumerator:

```
    def callback(monitor: T.Any, dc: T.Any, rect: T.Any, data: T.Any) -> int:
        info = MONITORINFOEXW()
        info.cbSize = ctypes.sizeof(MONITORINFOEXW)
        if ctypes.windll.user32.GetMonitorInfoW(monitor, ctypes.byref(info)):
            name = info.szDevice
        else:
            name = None

        dc2 = ctypes.windll.user32.GetDC(None)
        HORZSIZE = 4
        VERTSIZE = 6
        print(ctypes.windll.gdi32.GetDeviceCaps(dc2, HORZSIZE))
        print(ctypes.windll.gdi32.GetDeviceCaps(dc2, VERTSIZE))

        rct = rect.contents
        monitors.append(
            Monitor(
                x=rct.left,
                y=rct.top,
                width=rct.right - rct.left,
                height=rct.bottom - rct.top,
                name=name,
            )
        )
        return 1
```

However I have no clue how this works with more than one monitor. I got the key values from [here](https://github.com/SublimeText/Pywin32/blob/master/lib/x64/win32/lib/win32con.py).

**Comment by hhannine** [comment]
Update: Changing the code to
```
        dc2 = ctypes.windll.user32.GetDC(dc)
        HORZSIZE = 4
        VERTSIZE = 6
        print(ctypes.windll.gdi32.GetDeviceCaps(dc2, HORZSIZE))
        print(ctypes.windll.gdi32.GetDeviceCaps(dc2, VERTSIZE))
```
I got the same output as with the code I showed above so maybe this will work sensibly for every monitor as they get called by the EnumDisplayMonitors.

**Comment by hhannine** [comment]
I was able to figure it out and I've made a pull request with the changes at #24.

**Comment by rr-** [comment]
Thanks :+1: Merged and released as 0.6. Apple and Xinerama would need to go in separate pull requests.
